### ISSUE TITLE
Training forces help

### ISSUE THREAD

**Issue by Amitcuhp** [libAtoms/QUIP#617]
After doing every change in parameters, forces rmse are not coming below 100 meV/A
What can I do now? please help if possible. I am training cluster with 55 atoms.
    distance_2b
    order=2
    cutoff=8.0
    covariance_type=ard_se
    delta=0.001
    atom_sigma=0.8
    theta_uniform=1.0
    sparse_method=uniform
    add_species=T
    n_sparse=200
    :
    angle_3b
    cutoff=4.5
    covariance_type=ard_se
    delta=0.001
    atom_sigma=0.8
    theta_fac=0.5
    add_species=T
    n_sparse=600
    sparse_method=uniform
    :
    soap
    cutoff=8.0
    l_max=6
    n_max=12
    atom_sigma=0.8
    zeta=4 order=4
    delta=0.09
    cutoff_transition_width=1.0
    central_weight=1.0
    n_sparse=3000
    f0=0.0
    add_species=T
    covariance_type=dot_product
    sparse_method=cur_points

block default_sigma
     0.0001 
     0.01
     0.00000
     0.00000

**Comment by gabor1** [comment]
how many atoms are in your dataset?

**Comment by gabor1** [comment]
How many atoms are in your dataset? For clusters, it can be hard to get below 0.1 eV/A force accuracy  with GAP. Do you need better accuracy for your application? You might want to look at ACE or MACE.

**Comment by Amitcuhp** [comment]
there are 55 atoms in cluster.(one type atoms) And there are 5 different temperature ranges VASP data.
I think MACE  is good option . But I want to train in GAP. I am doing this training from a month, but it is not goiing below

**Comment by gabor1** [comment]
No I meant how much training data do you have. Can you post a graph of force error as a function of increasing training data? Also post a scatter plot of target force component vs predicted force component to see if you have outliers. Clusters are hard in general because the Fermi level fluctuates. Mace will give you better accuracy. Gábor CsányiProfessor of Molecular ModellingEngineering Laboratory, University of CambridgePembroke College CambridgePembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/On 24 Sep 2023, at 17:00, Amitcuhp ***@***.***> wrote:﻿
there are 55 atoms in cluster.(one type atoms) And there are 5 different temperature ranges VASP data.
I think MACE  is good option . But I want to train in GAP. I am doing this training from a month, but it is not goiing below

**Comment by Amitcuhp** [comment]
![Energy](https://github.com/libAtoms/QUIP/assets/97283086/ba78e8fa-f05d-41c5-9263-e8856c959e34)

![Force](https://github.com/libAtoms/QUIP/assets/97283086/ad23c305-b6c2-4f5b-81be-a3c472b66b64)

I have 8000 configurations. I am using 646 configurations for training.

**Comment by gabor1** [comment]
Remind me what element your clusters are made of? It is likely that the atom_sigma for soap is too high, try lower values, 0.6 and 0.4. your scatter plots look perfectly good, and your energy accuracy is very good too. Please produce a plot of force rmse Vs training data amount. Your total amount of training data is not that large.

**Comment by Amitcuhp** [comment]
Aluminum atoms are there.

**Comment by gabor1** [comment]
You can also try turbo_soap instead of soap, it almost always helps a bit. The parameters have different names but there is a good translation table in the turbo_soap GitHub repo.

**Comment by Amitcuhp** [comment]
Thank you very much for the information. I will plot force rmse Vs training data.and also increase amount of training data.
Also from your help, I am looking for MACE and turbo_soap. 

Again very thank you for your support.😀

**Comment by gabor1** [comment]
For aluminium the atom_sigma is definitely too high. Needs to be about 1/3 to 1/2 of the covalent radius. So try 0.4. at the same time reduce the soap cutoff from 8 to 6. The larger the cutoff the more training data you need.

**Comment by Amitcuhp** [comment]
![Screenshot from 2023-09-25 14-19-02](https://github.com/libAtoms/QUIP/assets/97283086/54a57d85-7639-4457-9622-0227dbc2e44f)
![Screenshot from 2023-09-25 14-25-01](https://github.com/libAtoms/QUIP/assets/97283086/1609ea5c-c5ee-4aaa-b140-7026fa6f9aae)
These are plots of absolute error VS configurations

**Comment by Amitcuhp** [comment]
distance_2b
    order=2
    cutoff=10.0
    covariance_type=ard_se
    delta=0.001
    atom_sigma=0.8
    theta_uniform=1.0
    sparse_method=uniform
    add_species=T
    n_sparse=20
    :
    angle_3b
    cutoff=3.5
    covariance_type=ard_se
    delta=0.001
    atom_sigma=0.8
    theta_fac=0.5
    add_species=T
    n_sparse=100
    sparse_method=uniform
    :
    soap
    cutoff=10.0
    l_max=6
    n_max=12
    atom_sigma=0.8
    zeta=4 order=4
    delta=0.09
    cutoff_transition_width=2.0
    central_weight=1.0
    n_sparse=8000
    f0=0.0
    add_species=T

These are inputs with best result. I have set atom_sigma 0.4 ,0.5, 0.6 : the error is incresing very much. Also cutoff radius I have tried error is minimum at this input parameters.

I don't know error can go below this by tunning the model or not. Can it ?

**Comment by gabor1** [comment]
Please use log y scale for error!!!

When you changed the atom_sigma did you change all of them or just the one for soap? It is the latter that I suggested.

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/


On 25 Sep 2023, at 10:57, Amitcuhp ***@***.***> wrote:

﻿

[Screenshot from 2023-09-25 14-19-02]<https://user-images.githubusercontent.com/97283086/270285496-54a57d85-7639-4457-9622-0227dbc2e44f.png>
[Screenshot from 2023-09-25 14-25-01]<https://user-images.githubusercontent.com/97283086/270285528-1609ea5c-c5ee-4aaa-b140-7026fa6f9aae.png>
These are plots of absolute error VS configurations

**Comment by Amitcuhp** [comment]
All of them. 
 Have I change only for soap?

**Comment by gabor1** [comment]
Ok, that’s not what I said, and it wouldn’t work very well. Keep the 2b and 3b at 0.8, but move the soap atom_sigma to 0.4. they mean slightly different things for the different descriptors, mathematically speaking.

And please plot the error log y scale.

-- Gábor

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/

On 25 Sep 2023, at 12:27, Amitcuhp ***@***.***> wrote:



All of them

**Comment by gabor1** [comment]
Also, I don’t quite understand your plots. What is EXACTLY on the x axis? What I want you to plot, is the RMSE on the train and test sets (log y scale) for DIFFERNT MODELS that are trained on DIFFERENT AMOUNTS of training data. Plot the RMSE as a function of the number of training data points used. You don’t need very many points, maybe 100,200,400,800,1600, 3200 training data structures (you said you used 646 before).

-- Gábor

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/

On 25 Sep 2023, at 14:43, Gabor Csanyi ***@***.***> wrote:

Ok, that’s not what I said, and it wouldn’t work very well. Keep the 2b and 3b at 0.8, but move the soap atom_sigma to 0.4. they mean slightly different things for the different descriptors, mathematically speaking.

And please plot the error log y scale.

-- Gábor

Gábor Csányi
Professor of Molecular Modelling
Engineering Laboratory, University of Cambridge
Pembroke College Cambridge

Pembroke College supports CARA. A Lifeline to Academics at Risk.  http://www.cara.ngo/

On 25 Sep 2023, at 12:27, Amitcuhp ***@***.***> wrote:



All of them

**Comment by Amitcuhp** [comment]
![Energy](https://github.com/libAtoms/QUIP/assets/97283086/1bdcd322-ecd4-4ea8-b165-6fffa59a27be)
![Force](https://github.com/libAtoms/QUIP/assets/97283086/e3aed277-9241-4cfb-a7aa-4e756bdd93ea)

I have reduces the errors to this.

**Comment by gabor1** [comment]
How did you manage this? It's better than I had dared to hold hope.

**Comment by Amitcuhp** [comment]
The problem is with VASP data. The SCF tag is not enabled. After doing that this is first run.

**Comment by gabor1** [comment]
So you didn't converge the SCFs at all?

**Comment by Amitcuhp** [comment]
We forgot to uncomment the tag.
Now it's working well. :)

**Comment by Amitcuhp** [comment]
I have tried MACE as suggested by you, it is working with CUDA which is very fast.
Giving very good result. 
Force RMSE is less than 10 meV/A.
